src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceRepositoryNameContributor.java

```diff
@@ -0,0 +1,30 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.cloudbees.jenkins.GitHubRepositoryName;
+import com.cloudbees.jenkins.GitHubRepositoryNameContributor;
+import hudson.Extension;
+import hudson.model.Item;
+import jenkins.branch.MultiBranchProject;
+
+import java.util.Collection;
+
+@Extension
+public class GitHubSCMSourceRepositoryNameContributor extends GitHubRepositoryNameContributor {
+
+    @Override
+    public void parseAssociatedNames(Item item, Collection<GitHubRepositoryName> result) {
```

Comment:
you should be able to write a JenkinsRule test for this method that verifies it obeys the contract as a minimum